src/modules/units/hooks/useDeleteUnits.tsx

```diff
@@ -0,0 +1,108 @@
+import DeleteIcon from '@mui/icons-material/Delete';
+import { Typography } from '@mui/material';
+import pluralize from 'pluralize';
+import { useCallback, useMemo, useState } from 'react';
+import ButtonTooltipContainer from '@/components/elements/ButtonTooltipContainer';
+import DeleteMutationButton from '@/modules/dataFetching/components/DeleteMutationButton';
+import DeleteMutationConfirmationDialog from '@/modules/dataFetching/components/DeleteMutationConfirmationDialog';
+import { evictUnitsQuery } from '@/modules/units/util';
+import {
+  DeleteUnitsDocument,
+  DeleteUnitsMutation,
+  DeleteUnitsMutationVariables,
+} from '@/types/gqlTypes';
+
+export const useDeleteUnits = ({
+  projectId,
+}: {
+  projectId: string;
+}): {
+  setUnitToDelete: (unitId: string) => void;
+  renderSingleDeleteDialog: () => JSX.Element;
+  renderBulkDeleteButton: (
+    unitIds: string[],
+    disabled?: boolean
+  ) => JSX.Element;
+} => {
+  const deleteDialogProps = useMemo(() => {
+    return {
+      idPath: 'deleteUnits.unitIds[0]',
```

Comment:
This is a bit unusual, I don't think we've used this component for deleting multiple records. Do we need to change the API at all? WiIl this work for retrieving and displaying validation errors?

Reply:
Oh I guess we were already doing this before